os/board/common/crashdump.c

```diff
@@ -208,6 +210,65 @@ static int get_ext_userfs_part(size_t *size, int *index)
 	return 0;
 }
 
+static int send_file(void)
+{
+	int i;
+	int fd;
+	int ret;
+	struct stat st;
+	off_t size;
+	char buf;
+	char filename[CONFIG_SMARTFS_MAXNAMLEN];
+	char char_filesize;
+
+	/* Send acknowledgement for USERFS FILE DUMP Handshake */
+	up_lowputc('A');
+
+	/* Receive the file name to be sent to the host */
+	for (i = 0; i < CONFIG_SMARTFS_MAXNAMLEN; i++) {
+		filename[i] = up_getc();
+	}
+
+	ret = stat(filename, &st);
+	if (ret != OK) {
+		/* We are not able to stat the file for some reason */
+		return -1;
+	}
+	size = st.st_size;
+
+	/* This Ack signal confirms the presence of the requested file */
+	//up_lowputc('A');
+
+	/* Send the size of the file we are dumping to the host so that it knows how much data is coming in
+	 * It is a 64 bit data type so has to be transmitted accordingly
+	 * This also acts as a confirmation of the presence of the file in itself.
+	 */
+	for (i = 0; i < 8; i++) {
+		char_filesize = (size >> ((8 - 1 - i) * 8));
+		up_lowputc(char_filesize);
+	}
+
+	fd = open(filename, O_RDOK);
+	if (fd < 0) {
+		//Unable to open file for transmitting data
+		return -EIO;
+	}
+
+	while (size) {
```

Comment:
You already know less I/O always improve performance.
```
char *buf = malloc(size...);
ret = read(fd, buf, size);
....
while (*size) {
   up_lowputc(*size++);
}
```

Reply:
I see crashdump.c does not use malloc at any place.
Also, are we really concerned about I/O time/performance when device is crashed? Our primary aim should be that we are able to extract the file/data correctly from the crashed device for analysis.

Reply:
@kishore-sn what is your opinion on this please?

Reply:
In the crashed state, we dont know what is the state of kernel. So, all kernel operations must be avoided as much as possible. So, let us avoid malloc. However, if reading one character at a time from the file causes too much overhead, then we could maintain a static buffer of sectorsize or blocksize. It is left to you to experiment with this and decide if it is required.